src/proto/single/resp.rs

```diff
@@ -59,6 +59,32 @@ pub fn read_json<R: BufRead, T: serde::de::DeserializeOwned>(r: R) -> Result<Opt
 
 // ----------------------------------------------
 
+#[cfg(feature = "ent")]
+pub fn read_bid<R: BufRead>(r: R) -> Result<String, Error> {
+    match read(r)? {
+        RawResponse::String(ref s) if s.is_empty() => Err(error::Protocol::BadType {
+            expected: "non-empty string representation of batch id",
+            received: "empty string".into(),
+        }
+        .into()),
+        RawResponse::String(ref s) => Ok(s.to_string()),
+        RawResponse::Blob(ref b) if b.is_empty() => Err(error::Protocol::BadType {
+            expected: "non-empty blob representation of batch id",
+            received: "empty blob".into(),
+        }
+        .into()),
+        RawResponse::Blob(ref b) => Ok(std::str::from_utf8(b)
+            .map_err(|_| error::Protocol::BadType {
+                expected: "valid blob representation of batch id",
+                received: "unprocessable blob".into(),
+            })?
+            .into()),
```

Comment:
Will the server ever actually send a batch ID as a blob? If not, we should error on it.

Reply:
Turned out it always sends it as a blob. It was just too tempting to include both options.